src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -466,11 +467,278 @@ type Moon {
+            System.Threading.Thread.Sleep(1000);
+
+            // Act
+            result = await httpClient.GetAsync($"{restPath}/Book");
+            string reloadRestContent = await result.Content.ReadAsStringAsync();
+            JsonElement reloadGQLContents = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                httpClient,
+                configProvider,
+                graphQLQueryName,
+                graphQLQuery);
+
+            // Assert
+            Assert.IsTrue(SqlTestHelper.JsonStringsDeepEqual(restBookContents, reloadRestContent));
+            SqlTestHelper.PerformTestEqualJsonStrings(BOOK_DBO_CONTENTS, reloadGQLContents.GetProperty("items").ToString());
+        }
+
+        /// <summary>
+        /// Start DAB with pre-formed config file, assert that requests sent to
+        /// the rest and gql endpoints result in result status OK. Hot-reload
+        /// the config and set rest and graphql enabled to false, and assert
+        /// that requests sent to the rest and gql endpoints result in result
+        /// status NotFound.
+        /// </summary>
+        /// <returns></returns>
+        [TestCategory(MSSQL_ENVIRONMENT)]
+        [TestMethod("Hot-reload rest and gql enabled.")]
+        public async Task HotReloadConfigRuntimeEnabledEndToEndTest()
+        {
+            // Arrange
+            string restEnabled = "false";
+            string gQLEnabled = "false";
+            string query = HOT_RELOAD_GQL_QUERY;
+            object payload =
+                new { query };
+
+            GenerateConfigFile(connectionString: $"{GetConnectionStringFromEnvironmentConfig(TestCategory.MSSQL)}");
+            TestServer server = new(Program.CreateWebHostBuilder(new string[] { "--ConfigFileName", HOT_RELOAD_CONFIG_FILE_NAME  }));    
+            HttpClient httpClient = server.CreateClient();
+            HttpRequestMessage request = new(HttpMethod.Post, "/graphQL")
+            {
+                Content = JsonContent.Create(payload)
+            };
+
+            // Act
+            HttpResponseMessage restResult = await httpClient.GetAsync("/rest/Book");
+            HttpResponseMessage gQLResult = await httpClient.SendAsync(request);
+
+            // Assert
+            Assert.AreEqual(HttpStatusCode.OK, restResult.StatusCode);
+            Assert.AreEqual(HttpStatusCode.OK, gQLResult.StatusCode);
+
+            // Arrange
+            request = new(HttpMethod.Post, "/graphQL")
+            {
+                Content = JsonContent.Create(payload)
+            };
+            GenerateConfigFile(
+                connectionString: $"{GetConnectionStringFromEnvironmentConfig(TestCategory.MSSQL)}",
+                restEnabled: restEnabled,
+                gQLEnabled: gQLEnabled);
+            System.Threading.Thread.Sleep(1000);
+
+            // Act
+            restResult = await httpClient.GetAsync($"rest/Book");
+            gQLResult = await httpClient.SendAsync(request);
```

Comment:
nit: The hot reload operation is the "Act" as it triggers the side effects you want to validate. The "Assert" section would include sending your requests to DAB's rest/graphql endpoints to ensure status is 404